### PR TITLE
Implemented ERC1155UriStorage

### LINKED ISSUE #486: Add more tests for ERC1155SetUri
Add more tests for:
- `src/dollar/ERC1155SetUri/ERC1155BurnableSetUri.sol`
- `src/dollar/ERC1155SetUri/ERC1155PausableSetUri.sol`
- `src/dollar/ERC1155SetUri/ERC1155SetUri.sol`

Test coverage right now:

| File                                               | % Lines           | % Statements      | % Branches        | % Funcs          |
|----------------------------------------------------|-------------------|-------------------|-------------------|------------------|
| src/dollar/ERC1155SetUri/ERC1155BurnableSetUri.sol | 0.00% (0/4)       | 0.00% (0/4)       | 0.00% (0/4)       | 0.00% (0/2)      |
| src/dollar/ERC1155SetUri/ERC1155PausableSetUri.sol | 100.00% (2/2)     | 100.00% (2/2)     | 50.00% (1/2)      | 100.00% (1/1)    |
| src/dollar/ERC1155SetUri/ERC1155SetUri.sol         | 56.52% (52/92)    | 54.39% (62/114)   | 25.00% (10/40)    | 66.67% (14/21)   |

### PR DESCRIPTION
**Pull request by FibrinLab**
Resolves #486

Ready for review

### CODE REVIEW

**Comment by FibrinLab** [comment]
@rndquu

File: packages/contracts/src/dollar/StakingShare.sol
Comment (rndquu) [inline packages/contracts/src/dollar/StakingShare.sol]: we already have this `_uri` property in the `ERC1155` contract so you can remove it

File: packages/contracts/src/dollar/StakingShare.sol
Comment (rndquu) [inline packages/contracts/src/dollar/StakingShare.sol]: This method returns `uri` from the `StakingShare` contract. If we call `setUri(1, "https://site.com/tokens/1.json")` then `uri(1)` will return `_uri` from the `StakingShare` contract, but it is expected to return `https://site.com/tokens/1.json`.

You can refactor the code with the following methods:
```
    function uri(
        uint256 tokenId
    ) public view virtual override(ERC1155, ERC1155URIStorage) returns (string memory) {
        return super.uri(tokenId);
    }

    function setBaseUri(string memory newUri) external onlyStakingManager {
        _setBaseURI(newUri);
    }

    function setUri(uint256 tokenId, string memory tokenUri) external onlyStakingManager {
        _setURI(tokenId, tokenUri);
    }

    function setUri(string memory tokenUri) external onlyStakingManager {
        _setURI(tokenUri);
    }
```

This way we have the ability to update URI for the `ERC1155` contract as well as URIs for single tokens which might be useful in the future.

So please update the code along with tests.

**Review by rndquu — CHANGES_REQUESTED** [review]

File: packages/contracts/src/dollar/StakingShare.sol
Comment (FibrinLab) [reply packages/contracts/src/dollar/StakingShare.sol]: done

File: packages/contracts/src/dollar/StakingShare.sol
Comment (FibrinLab) [reply packages/contracts/src/dollar/StakingShare.sol]: This is cool. Code updated

**Comment by FibrinLab** [comment]
@rndquu Thanks for the review. I made updates.

**Review by rndquu — APPROVED** [review]

File: packages/contracts/src/dollar/StakingShare.sol
Comment (0x4007) [inline packages/contracts/src/dollar/StakingShare.sol]:
```suggestion
    ERC1155UriStorage,
```

I noticed this was updated in @zgorizzo69 pull request the other week but I think that we should have this be `Uri` instead of `URI` to match our code style after changing `NFT` to `Nft` in identifiers. 

I understand that this is not the fault of this PR.

File: packages/contracts/test/dollar/StakingShare.t.sol
Comment (0x4007) [inline packages/contracts/test/dollar/StakingShare.t.sol]:
```suggestion
        "'Ubiquity Staking Share',"
```

File: packages/contracts/test/dollar/StakingShare.t.sol
Comment (0x4007) [inline packages/contracts/test/dollar/StakingShare.t.sol]:
```suggestion
        string memory stringTest = "{'name':'Staking Share','description':,"
```

**Comment by 0x4007** [comment]
You can merge after updating the test strings.

**Comment by zgorizzo69** [comment]
yep seems good to me too

**Comment by FibrinLab** [comment]
@pavlovcik @zgorizzo69 Conventional commit seems to fail after accepting the recommended changes. Any recommendations ?

**Comment by rndquu** [comment]
> @pavlovcik @zgorizzo69 Conventional commit seems to fail after accepting the recommended changes. Any recommendations ?

we have an issue for this case https://github.com/ubiquity/ubiquity-dollar/issues/590

for now I guess @pavlovcik will skip conventional commits check and merge the PR

**Comment by FibrinLab** [comment]
> > @pavlovcik @zgorizzo69 Conventional commit seems to fail after accepting the recommended changes. Any recommendations ?
> 
> 
> 
> we have an issue for this case https://github.com/ubiquity/ubiquity-dollar/issues/590
> 
> 
> 
> for now I guess @pavlovcik will skip conventional commits check and merge the PR

@rndquu Rather than ignoring commits. I could make a fresh PR with all CI passing. Thoughts?

**Comment by rndquu** [comment]
> > > @pavlovcik @zgorizzo69 Conventional commit seems to fail after accepting the recommended changes. Any recommendations ?
> > 
> > 
> > we have an issue for this case #590
> > for now I guess @pavlovcik will skip conventional commits check and merge the PR
> 
> @rndquu Rather than ignoring commits. I could make a fresh PR with all CI passing. Thoughts?

New PR will have to be reviewed again, let's wait for @pavlovcik to merge

**Comment by rndquu** [comment]
@FibrinLab hi, could you resolve the merge conflicts?

**Comment by FibrinLab** [comment]
> @FibrinLab hi, could you resolve the merge conflicts?

The conflict is still with Conventional Commit. I could only amend the last commit message. Any recommendations @rndquu

**Comment by rndquu** [comment]
> > @FibrinLab hi, could you resolve the merge conflicts?
> 
> The conflict is still with Conventional Commit. I could only amend the last commit message. Any recommendations @rndquu

There are conflicts in the code

```
This branch has conflicts that must be resolved

Conflicting files
packages/contracts/src/dollar/StakingShare.sol
packages/contracts/test/dollar/StakingShare.t.sol
```

**Comment by FibrinLab** [comment]
> ### This branch has conflicts that must be resolved
> Only those with [write access](https://docs.github.com/articles/what-are-the-different-access-permissions) to this repository can merge pull requests.
> #### Conflicting files
> `packages/contracts/src/dollar/StakingShare.sol`
> `packages/contracts/test/dollar/StakingShare.t.sol`

I just resolved the CC CI. Only @pavlovcik with write access can resolve. All the CI passing now. I don't have write access

**Comment by rndquu** [comment]
> > ### This branch has conflicts that must be resolved
> > Only those with [write access](https://docs.github.com/articles/what-are-the-different-access-permissions) to this repository can merge pull requests.
> > #### Conflicting files
> > `packages/contracts/src/dollar/StakingShare.sol`
> > `packages/contracts/test/dollar/StakingShare.t.sol`
> 
> I just resolved the CC CI. Only @pavlovcik with write access can resolve. All the CI passing now. I don't have write access

I don't get it :)

As far as I understand there are code conflicts in the following files:
- packages/contracts/src/dollar/StakingShare.sol
- packages/contracts/test/dollar/StakingShare.t.sol

So to resolve them you should:
1. merge the latest `development` branch into the `FibrinLab:bonding-curve` branch
2. fix conflicts
3. commit and push changes

Or you mean something else?

**Comment by FibrinLab** [comment]
> > > ### This branch has conflicts that must be resolved
> > > Only those with [write access](https://docs.github.com/articles/what-are-the-different-access-permissions) to this repository can merge pull requests.
> > > #### Conflicting files
> > > `packages/contracts/src/dollar/StakingShare.sol`
> > > `packages/contracts/test/dollar/StakingShare.t.sol`
> > 
> > 
> > I just resolved the CC CI. Only @pavlovcik with write access can resolve. All the CI passing now. I don't have write access
> 
> I don't get it :)
> 
> As far as I understand there are code conflicts in the following files:
> 
> * packages/contracts/src/dollar/StakingShare.sol
> * packages/contracts/test/dollar/StakingShare.t.sol
> 
> So to resolve them you should:
> 
> 1. merge the latest `development` branch into the `FibrinLab:bonding-curve` branch
> 2. fix conflicts
> 3. commit and push changes
> 
> Or you mean something else?

@rndquu Thanks for pointing me in the right direction. I noticed the repo has switched to Diamond using facets. I made updates to the code and tests. Thanks for the review

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (rndquu) [inline packages/contracts/src/dollar/core/StakingShare.sol:11]: Here you should inherit from `ERC1155Ubiquity , ERC1155URIStorage `

The `ERC1155Ubiquity` contract is `ERC1155Pausable` and `ERC1155Burnable` under the hood. It also has access control. So if you inherit from `ERC1155Ubiquity` then you can delete:
- https://github.com/ubiquity/ubiquity-dollar/blob/2926ae730aad94c6090bd016ccc9927537f39a9b/packages/contracts/src/dollar/core/StakingShare.sol#L38
- https://github.com/ubiquity/ubiquity-dollar/blob/2926ae730aad94c6090bd016ccc9927537f39a9b/packages/contracts/src/dollar/core/StakingShare.sol#L72
- https://github.com/ubiquity/ubiquity-dollar/blob/2926ae730aad94c6090bd016ccc9927537f39a9b/packages/contracts/src/dollar/core/StakingShare.sol#L131
- https://github.com/ubiquity/ubiquity-dollar/blob/2926ae730aad94c6090bd016ccc9927537f39a9b/packages/contracts/src/dollar/core/StakingShare.sol#L141
- https://github.com/ubiquity/ubiquity-dollar/blob/2926ae730aad94c6090bd016ccc9927537f39a9b/packages/contracts/src/dollar/core/StakingShare.sol#L141
- https://github.com/ubiquity/ubiquity-dollar/blob/2926ae730aad94c6090bd016ccc9927537f39a9b/packages/contracts/src/dollar/core/StakingShare.sol#L162
- https://github.com/ubiquity/ubiquity-dollar/blob/2926ae730aad94c6090bd016ccc9927537f39a9b/packages/contracts/src/dollar/core/StakingShare.sol#L197
- https://github.com/ubiquity/ubiquity-dollar/blob/2926ae730aad94c6090bd016ccc9927537f39a9b/packages/contracts/src/dollar/core/StakingShare.sol#L203

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (rndquu) [inline packages/contracts/src/dollar/core/StakingShare.sol]: Inherit from the `ERC1155Ubiquity` contract and keep the `override`

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (rndquu) [inline packages/contracts/src/dollar/core/StakingShare.sol]: Inherit from the `ERC1155Ubiquity` contract and keep the `override`

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (rndquu) [inline packages/contracts/src/dollar/core/StakingShare.sol]: Inherit from the `ERC1155Ubiquity` contract and keep the `override`

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (rndquu) [inline packages/contracts/src/dollar/core/StakingShare.sol]: Inherit from the `ERC1155Ubiquity` contract and keep the `override`

**Review by rndquu — CHANGES_REQUESTED** [review]

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (FibrinLab) [reply packages/contracts/src/dollar/core/StakingShare.sol:11]: Thanks for the review. Some functions are requesting to be overwritten so I let them there for your review. ERC1155URIStorage also overwrites those same functions.

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (FibrinLab) [reply packages/contracts/src/dollar/core/StakingShare.sol]: done

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (FibrinLab) [reply packages/contracts/src/dollar/core/StakingShare.sol]: done

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (FibrinLab) [reply packages/contracts/src/dollar/core/StakingShare.sol]: done

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (FibrinLab) [reply packages/contracts/src/dollar/core/StakingShare.sol]: done

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (rndquu) [inline packages/contracts/src/dollar/core/StakingShare.sol]: Why was this method removed? I think we should keep it in order to set URI for `ERC1155` contract

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (rndquu) [inline packages/contracts/src/dollar/core/StakingShare.sol:125]: We should remove this method because it is already implemented here https://github.com/ubiquity/ubiquity-dollar/blob/4b1e1516df5bc41b68c0fcf9083743122fb32933/packages/contracts/src/dollar/core/ERC1155Ubiquity.sol#L135

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (rndquu) [inline packages/contracts/src/dollar/core/StakingShare.sol:159]: We should remove this method because it is already implemented here https://github.com/ubiquity/ubiquity-dollar/blob/4b1e1516df5bc41b68c0fcf9083743122fb32933/packages/contracts/src/dollar/core/ERC1155Ubiquity.sol#L149

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (rndquu) [inline packages/contracts/src/dollar/core/StakingShare.sol:169]: We should remove this method because it is already implemented here https://github.com/ubiquity/ubiquity-dollar/blob/4b1e1516df5bc41b68c0fcf9083743122fb32933/packages/contracts/src/dollar/core/ERC1155Ubiquity.sol#L185

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (rndquu) [inline packages/contracts/src/dollar/core/StakingShare.sol:177]: We should remove this method because it is already implemeted here https://github.com/ubiquity/ubiquity-dollar/blob/4b1e1516df5bc41b68c0fcf9083743122fb32933/packages/contracts/src/dollar/core/ERC1155Ubiquity.sol#L196

**Review by rndquu — CHANGES_REQUESTED** [review]

**Comment by FibrinLab** [comment]
Screenshot 2023-03-20 at 16 41 38
@rndquu Have a look at this. Those functions need to be overridden

**Comment by rndquu** [comment]
> Screenshot 2023-03-20 at 16 41 38
> 
> @rndquu Have a look at this. Those functions need to be overridden

Yes, you're right. Then keep those overridden methods but don't duplicate method implementations, simply call parent methods. So the following lines should be removed:
- https://github.com/ubiquity/ubiquity-dollar/blob/4b1e1516df5bc41b68c0fcf9083743122fb32933/packages/contracts/src/dollar/core/StakingShare.sol#L131
- https://github.com/ubiquity/ubiquity-dollar/blob/4b1e1516df5bc41b68c0fcf9083743122fb32933/packages/contracts/src/dollar/core/StakingShare.sol#L166
- https://github.com/ubiquity/ubiquity-dollar/blob/4b1e1516df5bc41b68c0fcf9083743122fb32933/packages/contracts/src/dollar/core/StakingShare.sol#L175

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (FibrinLab) [reply packages/contracts/src/dollar/core/StakingShare.sol]: Already implemented in ERC1155Ubiquity

**Comment by FibrinLab** [comment]
@rndquu Thanks for the review. All corrections implemented

**Review by rndquu — APPROVED** [review]

**Comment by zgorizzo69** [comment]
here is the storage layout of stakingShare
```
| Name               | Type                                            | Slot | Offset | Bytes | Contract                                      |
|--------------------|-------------------------------------------------|------|--------|-------|-----------------------------------------------|
| _balances          | mapping(uint256 => mapping(address => uint256)) | 0    | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
| _operatorApprovals | mapping(address => mapping(address => bool))    | 1    | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
| _uri               | string                                          | 2    | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
| _paused            | bool                                            | 3    | 0      | 1     | src/dollar/core/StakingShare.sol:StakingShare |
| accessCtrl         | contract IAccessControl                         | 3    | 1      | 20    | src/dollar/core/StakingShare.sol:StakingShare |
| _holderBalances    | mapping(address => uint256[])                   | 4    | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
| _totalSupply       | uint256                                         | 5    | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
| _baseURI           | string                                          | 6    | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
| _tokenURIs         | mapping(uint256 => string)                      | 7    | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
| _holderBalances    | mapping(address => uint256[])                   | 8    | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
| _stakes            | mapping(uint256 => struct StakingShare.Stake)   | 9    | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
| _totalLP           | uint256                                         | 10   | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
| _totalSupply       | uint256                                         | 11   | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
```
note that we have `_uri` and `_tokenURIs`
`_uri` is only set in the constructor but when we are calling stakingShare.uri(tokenId) we actually get data from `_tokenURIs` so should we keep it ? should we add a getter for `_uri ` ?
 moreover there is also `_baseURI` which is concatenate with the tokenURI but there is no test about that ...

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (zgorizzo69) [inline packages/contracts/src/dollar/core/StakingShare.sol:217]: no test for that

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (zgorizzo69) [inline packages/contracts/src/dollar/core/StakingShare.sol:191]: should we have a getter for _uri ?

**Review by zgorizzo69 — CHANGES_REQUESTED** [review]

**Comment by FibrinLab** [comment]
> here is the storage layout of stakingShare
> 
> ```
> 
> | Name               | Type                                            | Slot | Offset | Bytes | Contract                                      |
> 
> |--------------------|-------------------------------------------------|------|--------|-------|-----------------------------------------------|
> 
> | _balances          | mapping(uint256 => mapping(address => uint256)) | 0    | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
> 
> | _operatorApprovals | mapping(address => mapping(address => bool))    | 1    | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
> 
> | _uri               | string                                          | 2    | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
> 
> | _paused            | bool                                            | 3    | 0      | 1     | src/dollar/core/StakingShare.sol:StakingShare |
> 
> | accessCtrl         | contract IAccessControl                         | 3    | 1      | 20    | src/dollar/core/StakingShare.sol:StakingShare |
> 
> | _holderBalances    | mapping(address => uint256[])                   | 4    | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
> 
> | _totalSupply       | uint256                                         | 5    | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
> 
> | _baseURI           | string                                          | 6    | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
> 
> | _tokenURIs         | mapping(uint256 => string)                      | 7    | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
> 
> | _holderBalances    | mapping(address => uint256[])                   | 8    | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
> 
> | _stakes            | mapping(uint256 => struct StakingShare.Stake)   | 9    | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
> 
> | _totalLP           | uint256                                         | 10   | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
> 
> | _totalSupply       | uint256                                         | 11   | 0      | 32    | src/dollar/core/StakingShare.sol:StakingShare |
> 
> ```
> 
> note that we have `_uri` and `_tokenURIs`
> 
> `_uri` is only set in the constructor but when we are calling stakingShare.uri(tokenId) we actually get data from `_tokenURIs` so should we keep it ? should we add a getter for `_uri ` ?
> 
>  moreover there is also `_baseURI` which is concatenate with the tokenURI but there is no test about that ...

@rndquu

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (FibrinLab) [reply packages/contracts/src/dollar/core/StakingShare.sol:217]: fixed

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (FibrinLab) [reply packages/contracts/src/dollar/core/StakingShare.sol:191]: Not sure its important. @rndquu whats your opinion.

Comment (0x4007) [reply packages/contracts/src/dollar/core/StakingShare.sol:191]: If theres a question around it you should just add it. I think theres almost no downside to adding getters.

**Review by zgorizzo69 — APPROVED** [review]

**Comment by FibrinLab** [comment]
@rndquu @zgorizzo69 CI now passing

File: packages/contracts/src/dollar/core/StakingShare.sol
Comment (rndquu) [reply packages/contracts/src/dollar/core/StakingShare.sol:191]: @zgorizzo69 

This is how URI getter works: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/extensions/ERC1155URIStorage.sol#L46

So if URI is empty for a particular token id then return this URI https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/ERC1155.sol#L30 else return this URI https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/extensions/ERC1155URIStorage.sol#L22

So we don't need a separate getter for `ERC1155._uri` because we already have it here https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/extensions/ERC1155URIStorage.sol#L46

Comment (rndquu) [reply packages/contracts/src/dollar/core/StakingShare.sol:191]: > should we have a getter for _uri ?

You mean can't we delete it? Both `ERC1155 .sol` and `ERC1155URIStorage.sol` have the `uri()` method so we must override it

Comment (zgorizzo69) [reply packages/contracts/src/dollar/core/StakingShare.sol:191]: I guess you are right this part was confusing me 
```
  function uri(uint256) public view virtual override returns (string memory) {
        return _uri;
    }
```
But yeah no need for a getter then

File: packages/contracts/test/dollar/core/StakingShare.t.sol
Comment (0x4007) [inline packages/contracts/test/dollar/core/StakingShare.t.sol]:
```suggestion
        "'Ubiquity Staking Share',"
```

File: packages/contracts/test/dollar/core/StakingShare.t.sol
Comment (0x4007) [inline packages/contracts/test/dollar/core/StakingShare.t.sol]:
```suggestion
        string memory stringTest = "{'name':'Staking Share','description':,"
```

File: packages/contracts/test/dollar/core/StakingShare.t.sol
Comment (0x4007) [inline packages/contracts/test/dollar/core/StakingShare.t.sol]:
```suggestion
        string memory stringTest = "{'name':'Staking Share','description':,"
```

File: packages/contracts/test/dollar/core/StakingShare.t.sol
Comment (0x4007) [inline packages/contracts/test/dollar/core/StakingShare.t.sol]:
```suggestion
        "'Ubiquity Staking Share',"
```

Comment (0x4007) [reply packages/contracts/test/dollar/core/StakingShare.t.sol]: I thought we banned `Bonding` as a word in cspell

**Comment by 0x4007** [comment]
If the others approved of the logic then I trust thats all good. I just requested small adjustments related to the branding of the token. Once those are fixed you can merge it in!

**Comment by FibrinLab** [comment]
> If the others approved of the logic then I trust thats all good. I just requested small adjustments related to the branding of the token. Once those are fixed you can merge it in!

I just accepted the changes. Good to merge @pavlovcik

**Comment by FibrinLab** [comment]
@pavlovcik I think auto merge was disabled.

**Comment by FibrinLab** [comment]
@pavlovcik @rndquu Auto merge is disabled

**Comment by rndquu** [comment]
> @pavlovcik @rndquu Auto merge is disabled

Yes, auto merge was disabled, let's wait for @pavlovcik to merge

**Comment by 0x4007** [comment]
I see:

### This branch cannot be rebased due to conflicts

Those presumably must be addressed before GitHub will merge it.

**Comment by FibrinLab** [comment]
> I see:
> 
> ### This branch cannot be rebased due to conflicts
> Those presumably must be addressed before GitHub will merge it.
Screenshot 2023-03-30 at 09 00 @pavlovcik  src=
@pavlovcik I cant see to find any conflicts here

**Comment by FibrinLab** [comment]
> I see:
> 
> ### This branch cannot be rebased due to conflicts
> Those presumably must be addressed before GitHub will merge it.

Screenshot 2023-03-30 at 09 00 23

@pavlovcik I can seem to find this on my end. Recommendations?

**Comment by rndquu** [comment]
> I see:
> 
> ### This branch cannot be rebased due to conflicts
> Those presumably must be addressed before GitHub will merge it.

[
Screenshot 2023-03-31 at 14 19 14
](url)

Try "Create a merge commit" or "Squash and merge". The "Rebase and merge" option somehow has conflicts although there are no conflicts in the code.

**Pull request merged**
